packages/nodes-base/nodes/ExecuteWorkflow/ExecuteWorkflowTrigger/ExecuteWorkflowTrigger.node.ts

```diff
@@ -0,0 +1,206 @@
+			color: '#ff6d5a',
+		},
+		inputs: [],
+		outputs: [NodeConnectionType.Main],
+		hints: [
+			{
+				message:
+					'You need to define your input fields explicitly. Otherwise the parent cannot provide data and you will not receive input data.',
+				// This condition checks if we have no input fields, which gets a bit awkward:
+				// For WORKFLOW_INPUTS: keys() only contains `VALUES` if at least one value is provided
+				// For JSON_EXAMPLE: We remove all whitespace and check if we're left with an empty object. Note that we already error if the example is not valid JSON
+				displayCondition:
+					`={{$parameter['${INPUT_SOURCE}'] === '${WORKFLOW_INPUTS}' && !$parameter['${WORKFLOW_INPUTS}'].keys().length ` +
+					`|| $parameter['${INPUT_SOURCE}'] === '${JSON_EXAMPLE}' && $parameter['${JSON_EXAMPLE}'].toString().replaceAll(' ', '').replaceAll('\\n', '') === '{}' }}`,
+				whenToDisplay: 'always',
+				location: 'ndv',
+			},
+		],
+		properties: [
+			{
+				displayName: 'Events',
+				name: 'events',
+				type: 'hidden',
+				noDataExpression: true,
+				options: [
+					{
+						name: 'Workflow Call',
+						value: 'worklfow_call',
+						description: 'When called by another workflow using Execute Workflow Trigger',
+						action: 'When Called by Another Workflow',
+					},
+				],
+				default: 'worklfow_call',
+			},
+			{
+				displayName: 'Input Source',
+				name: INPUT_SOURCE,
+				type: 'options',
+				options: [
+					{
+						name: 'Using Fields Below',
+						value: WORKFLOW_INPUTS,
+						description: 'Provide via UI',
+					},
+					{
+						name: 'Using JSON Example',
+						value: JSON_EXAMPLE,
+						description: 'Infer JSON schema via JSON example output',
+					},
+				],
+				default: WORKFLOW_INPUTS,
+				noDataExpression: true,
+				displayOptions: {
+					show: { '@version': [{ _cnd: { gte: 1.1 } }] },
+				},
+			},
+			{
+				displayName:
+					'Provide an example object to infer fields and their types.<br>To allow any type for a given field, set the value to null.',
+				name: `${JSON_EXAMPLE}_notice`,
+				type: 'notice',
+				default: '',
+				displayOptions: {
+					show: { '@version': [{ _cnd: { gte: 1.1 } }], inputSource: [JSON_EXAMPLE] },
+				},
+			},
+			{
+				displayName: 'JSON Example',
+				name: JSON_EXAMPLE,
+				type: 'json',
+				default: JSON.stringify(
+					{
+						aField: 'a string',
+						aNumber: 123,
+						thisFieldAcceptsAnyType: null,
+						anArray: [],
+					},
+					null,
+					2,
+				),
+				noDataExpression: true,
+				displayOptions: {
+					show: { '@version': [{ _cnd: { gte: 1.1 } }], inputSource: [JSON_EXAMPLE] },
+				},
+			},
+			{
+				displayName: 'Workflow Inputs',
+				name: WORKFLOW_INPUTS,
+				placeholder: 'Add Field',
+				type: 'fixedCollection',
+				description:
+					'Define expected input fields. If no inputs are provided, all data from the calling workflow will be passed through.',
+				typeOptions: {
+					multipleValues: true,
+					sortable: true,
```

Comment:
Something to bear in mind while testing: Do we handle this gracefully or does this risk not being updated? We can drop this if it's an issue